src/backend/joanie/edx_imports/tasks/enrollments.py

```diff
@@ -0,0 +1,130 @@
+        batch_count += 1
+        start = current_enrollment_index + offset
+        stop = current_enrollment_index + batch_size
+        if limit:
+            stop = min(stop, limit)
+        import_enrollments_batch_task.delay(start=start, stop=stop, dry_run=dry_run)
+    logger.info("%s import enrollments tasks launched", batch_count)
+
+
+@app.task(bind=True)
+def import_enrollments_batch_task(self, **kwargs):
+    """
+    Task to import enrollments from the OpenEdx database to the Joanie database.
+    """
+    logger.info("Starting Celery task, importing enrollments...")
+    try:
+        report = import_enrollments_batch(**kwargs)
+    except Exception as e:
+        logger.exception(e)
+        raise self.retry(exc=e) from e
+    logger.info("Done executing Celery importing enrollments task...")
+    return report
+
+
+def import_enrollments_batch(start, stop, dry_run=False):
+    """Batch import enrollments from OpenEdx student_course_enrollment"""
+    db = OpenEdxDB()
+    report = {
+        "enrollments": {
+            "created": 0,
+            "updated": 0,
+            "errors": 0,
+        }
+    }
+    enrollments = db.get_enrollments(start, stop)
+    enrollments_to_create = []
+    enrollments_to_update = []
+
+    for edx_enrollment in enrollments:
+        try:
+            course_run = models.CourseRun.objects.only("pk").get(
+                resource_link__icontains=edx_enrollment.course_id
+            )
+        except models.CourseRun.DoesNotExist:
+            report["enrollments"]["errors"] += 1
+            logger.error("No CourseRun found for %s", edx_enrollment.course_id)
+            continue
+
+        user_name = edx_enrollment.user.username
+        try:
+            user = models.User.objects.only("pk").get(username=user_name)
+        except models.User.DoesNotExist:
+            report["enrollments"]["errors"] += 1
+            logger.error("No User found for %s", user_name)
+            continue
+
+        try:
+            enrollment = models.Enrollment.objects.get(course_run=course_run, user=user)
+            enrollment.is_active = edx_enrollment.is_active
+            enrollment.created_on = make_date_aware(edx_enrollment.created)
+            enrollments_to_update.append(enrollment)
+        except models.Enrollment.DoesNotExist:
+            enrollments_to_create.append(
+                models.Enrollment(
+                    course_run=course_run,
+                    user=user,
+                    is_active=edx_enrollment.is_active,
+                    created_on=make_date_aware(edx_enrollment.created),
+                )
+            )
+
+    import_string = "%s enrollments created, %s updated, %s errors"
+    if not dry_run:
+        enrollment_created_on_field = models.Enrollment._meta.get_field("created_on")
+        enrollment_created_on_field.auto_now_add = False
+        enrollment_created_on_field.editable = True
```

Comment:
I wonder if we should not do that for other model (Organization, Users, Course, Course Run ?)

Reply:
Why not, yes

Reply:
It is already done for course and course run, but this data doesn't exists in Open edX for universities and users.

Reply:
Ok so it looks ok like that : )